src/Endpoint/Partition.php

```diff
@@ -287,12 +287,18 @@ private function getVariant(array $options, array $data)
     {
         $variantTags = [];
         if (isset($options['use_fips_endpoint'])) {
-            if ($options['use_fips_endpoint']->isUseFipsEndpoint()) {
+            $useFips = $options['use_fips_endpoint'];
+            if (is_bool($useFips)) {
+                $useFips && $variantTags[] = 'fips';
+            } elseif ($useFips->isUseFipsEndpoint()) {
```

Comment:
This is typechecked when we initialize the client, right?  So it'll only be a bool or a config option that contains a method isUseFipsEndpoint?

Reply:
Yeah, it's typechecked for the top-level config option, but if someone instantiates a `PartitionEndpointProvider`, they have to also instantiate a fips and dualstack config to pass in as options.  This allows `'use_fips_endpoint` => true` (same for dualstack) to be passed in when instantiating the provider.